src/commands/config/setCluster.ts

```diff
@@ -0,0 +1,235 @@
+        // eslint-disable-next-line require-atomic-updates
+        server.name = selected.name;
+      }
+    } else {
+      selected = clusterOptions.find(c => c.cluster?.name === server.name)
+        ?.cluster;
+      if (!selected) {
+        const choices = clusterOptions
+          .filter(o => !o.debug || argv.debug)
+          .map(o => o.cluster?.name)
+          .filter(n => n);
+        throw new Error(
+          `Cluster not found. Valid choices are: ${choices.join(", ")}`
+        );
+      }
+    }
+
+    if (selected) {
+      if (!server.endpoint) {
+        // eslint-disable-next-line require-atomic-updates
+        server.endpoint = selected.endpoint;
+      }
+      if (!server.apiKey) {
+        // eslint-disable-next-line require-atomic-updates
+        server.apiKey = selected.apiKey;
+      }
+    }
+  }
+
+  if (!server.name) {
+    prompts.push(namePrompt);
+  }
+
+  if (server.endpoint) {
+    console.log(
+      chalk`Setup cluster API endpoint as {green ${server.endpoint}}.`
+    );
+  } else {
+    prompts.push(urlPrompt);
+  }
+
+  if (!server.apiKey) {
+    prompts.push(apiKeyPrompt);
+  }
+
+  if (prompts.length === 0) {
+    return Promise.resolve(server);
+  }
+
+  return inquirer.prompt(prompts).then(answers => {
+    return {
+      ...server,
+      ...answers,
+    };
+  });
+}
+
+async function run(argv: Arguments) {
+  const { endpoint, apiKey, name: clusterName } = await askClusterServer(argv);
+
+  await cliContainer.container.configure({
+    endpoint,
+    apiKey,
+  });
+  const { env } = await cliContainer.getClusterConfig();
+
+  const config = argv.context.skycliConfig!;
+  config.clusters = config.clusters ?? [];
+  config.contexts = config.contexts ?? [];
+
+  const clusterConfig: ClusterConfig = { env, endpoint, api_key: apiKey };
+  const cluster = config.clusters.find(c => c.name === clusterName);
+  if (!cluster) {
+    config.clusters.push({ name: clusterName, cluster: clusterConfig });
+  } else {
+    cluster.cluster = clusterConfig;
+  }
+
+  const context = config.contexts.find(c => c.context.cluster === clusterName);
+  if (!context) {
+    config.contexts.push({
+      name: clusterName,
+      context: { cluster: clusterName, user: clusterName },
+    });
+  } else {
+    context.context.cluster = clusterName;
+  }
+
+  if (!config.current_context) {
+    config.current_context = clusterName;
+    console.log(chalk`Current context set to {green ${clusterName}}.`);
+  } else {
+    console.log(chalk`Context {green ${clusterName}} added.`);
+  }
+  save(config, "global");
+
+  console.log(chalk`Connected to Skygear cluster at {green ${endpoint}}.`);
+}
+
+export default createCommand({
+  builder: yargs => {
+    return yargs
+      .option("endpoint", {
+        desc: "Cluster API endpoint.",
+        type: "string",
+      })
+      .option("api-key", {
+        desc: "Cluster API key.",
+        type: "string",
+      })
+      .option("cluster", {
+        desc: "Cluster type.",
```

Comment:
This description is misleading. Its effect is to provide both the name of the cluster and the context.